aosp_diff/preliminary/frameworks/av/07_0007-Fix-UAF-in-clearkey-service-s-MemoryFileSystem.patch

```diff
@@ -0,0 +1,212 @@
+         if (requestString.find(kOfflineLicense) != std::string::npos) {
+             std::string emptyResponse;
+             std::string keySetIdString(keySetId.begin(), keySetId.end());
++            Mutex::Autolock lock(mFileHandleLock);
+             if (!mFileHandle.StoreLicense(keySetIdString,
+                     DeviceFiles::kLicenseStateReleasing,
+                     emptyResponse)) {
+@@ -335,6 +336,7 @@ bool DrmPlugin::makeKeySetId(std::string* keySetId) {
+         }
+         *keySetId = kKeySetIdPrefix + ByteArrayToHexString(
+                 reinterpret_cast<const uint8_t*>(randomData.data()), randomData.size());
++        Mutex::Autolock lock(mFileHandleLock);
+         if (mFileHandle.LicenseExists(*keySetId)) {
+             // collision, regenerate
+             ALOGV("Retry generating KeySetId");
+@@ -392,6 +394,7 @@ Return<void> DrmPlugin::provideKeyResponse(
+     if (status == Status::OK) {
+         if (isOfflineLicense) {
+             if (isRelease) {
++                Mutex::Autolock lock(mFileHandleLock);
+                 mFileHandle.DeleteLicense(keySetId);
+                 mSessionLibrary->destroySession(session);
+             } else {
+@@ -400,6 +403,7 @@ Return<void> DrmPlugin::provideKeyResponse(
+                     return Void();
+                 }
+
++                Mutex::Autolock lock(mFileHandleLock);
+                 bool ok = mFileHandle.StoreLicense(
+                         keySetId,
+                         DeviceFiles::kLicenseStateActive,
+@@ -454,6 +458,7 @@ Return<Status> DrmPlugin::restoreKeys(
+         DeviceFiles::LicenseState licenseState;
+         std::string offlineLicense;
+         Status status = Status::OK;
++        Mutex::Autolock lock(mFileHandleLock);
+         if (!mFileHandle.RetrieveLicense(std::string(keySetId.begin(), keySetId.end()),
+                 &licenseState, &offlineLicense)) {
+             ALOGE("Failed to restore offline license");
+@@ -705,6 +710,8 @@ Return<void> DrmPlugin::getMetrics(getMetrics_cb _hidl_cb) {
+ }
+
+ Return<void> DrmPlugin::getOfflineLicenseKeySetIds(getOfflineLicenseKeySetIds_cb _hidl_cb) {
++    Mutex::Autolock lock(mFileHandleLock);
++
+     std::vector<std::string> licenseNames = mFileHandle.ListLicenses();
+     std::vector<KeySetId> keySetIds;
+     if (mMockError != Status_V1_2::OK) {
+@@ -725,6 +732,7 @@ Return<Status> DrmPlugin::removeOfflineLicense(const KeySetId& keySetId) {
+         return toStatus_1_0(mMockError);
+     }
+     std::string licenseName(keySetId.begin(), keySetId.end());
++    Mutex::Autolock lock(mFileHandleLock);
+     if (mFileHandle.DeleteLicense(licenseName)) {
+         return Status::OK;
+     }
+@@ -733,6 +741,8 @@ Return<Status> DrmPlugin::removeOfflineLicense(const KeySetId& keySetId) {
+
+ Return<void> DrmPlugin::getOfflineLicenseState(const KeySetId& keySetId,
+         getOfflineLicenseState_cb _hidl_cb) {
++    Mutex::Autolock lock(mFileHandleLock);
++
```

Comment:
extra line.

Reply:
> extra line.

This extra line is present in Google patch also.